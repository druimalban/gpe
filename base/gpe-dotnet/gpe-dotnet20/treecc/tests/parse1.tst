// test good node definitions

%node expression %abstract %typedef =
{
	%nocreate type_code type = {int_type};
}

%node binary expression %abstract =
{
	expression *expr1;
	expression *expr2;
}

%node unary expression %abstract =
{
	;
	expression *expr;
	;
}

%node intnum expression =
{
	int num;
}

%node plus binary
%node minus binary
%node multiply binary
%node divide binary
%node negate unary

%node power binary = {;;}
