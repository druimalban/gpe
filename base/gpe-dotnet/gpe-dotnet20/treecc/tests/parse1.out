line 3: %node expression no_parent 6
line 5: %field type type_code int_type 1
line 8: %node binary expression 2
line 10: %field expr1 expression * no_value 0
line 11: %field expr2 expression * no_value 0
line 14: %node unary expression 2
line 17: %field expr expression * no_value 0
line 21: %node intnum expression 0
line 23: %field num int no_value 0
line 26: %node plus binary 0
line 27: %node minus binary 0
line 28: %node multiply binary 0
line 29: %node divide binary 0
line 30: %node negate unary 0
line 32: %node power binary 0
